src/playdate/graphics.nim

```diff
@@ -243,6 +243,30 @@ proc getDisplayFrame*(this: ptr PlaydateGraphics): DisplayFrame =
     privateAccess(PlaydateGraphics)
     return cast[DisplayFrame](this.getDisplayFrame()) # Who should manage this memory? Not clear. Not auto-managed right now.
 
+proc frameIndex(x, y: SomeInteger): int = y * LCD_ROWSIZE + x div 8
+    ## Returns the index of a coordinate within a DisplayFrame
+
+proc frameBit(x: SomeInteger): uint8 = 1'u8 shl uint8(7 - (x mod 8))
+
+proc `[]`*(frame: DisplayFrame, x, y: SomeInteger): LCDSolidColor =
```

Comment:
Using generic types like SomeInteger is fine, but I think it would be better to have them as a separate PR that changes the signature all around the module, where feasible.

Also, there's a small issue in VSCode using this type:
![Code 2023-01-31 at 12 57 16](https://user-images.githubusercontent.com/19392104/215753554-5740a810-6dab-465b-8c82-e6a279607dae.png)

That \*missing parameters\* in the proc signature appears when using generic types.

Reply:
I'm fine with removing them for now